Scripts/Runtime/Entities/Util/WorldUtil.cs

```diff
@@ -0,0 +1,230 @@
+        /// making it impractical to group with the other worlds. There shouldn't be any expensive work in that command buffer anyway.
+        /// </remarks>
+        public static void OptimizeWorldForMultiWorldInCurrentPlayerLoop(World world)
+        {
+            AddCustomPhasesToCurrentPlayerLoop();
+            AddTopLevelGroupsToCurrentPlayerLoop(world, s_MultiWorldTopLevelGroupTypes);
+
+            MoveSystemFromToGroup<EndInitializationEntityCommandBufferSystem, InitializationSystemGroup, EndInitializationCommandBufferSystemGroup>(world);
+            MoveSystemFromToGroup<EndSimulationEntityCommandBufferSystem, SimulationSystemGroup, EndSimulationCommandBufferSystemGroup>(world);
+
+            // Suppress the logging during sort so we don't see complaints about systems that position themselves based on
+            // the systems that we're moving. So far, the configured above are at the end of the group and don't cause issues
+            // aside from the warnings. (dependent systems end up in the right place anyway)
+            try
+            {
+#if LOG_VERBOSE
+                UnityEngine.Debug.LogWarning($"Sorting {nameof(PlayerLoopSystem)}s. Warning expected with systems trying to position against ${nameof(EndSimulationEntityCommandBufferSystem)}");
+#else
+                Log.SupressLogging = true;
+#endif
+                SortAllGroupsInWorld(world);
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+            finally
+            {
+                Log.SupressLogging = false;
+            }
+        }
+
+        /// <summary>
+        /// Moves a system from one group to another.
+        /// </summary>
+        /// <typeparam name="System">The system type to move</typeparam>
+        /// <typeparam name="SrcGroup">The system's existing group.</typeparam>
+        /// <typeparam name="DestGroup">The system's destination group.</typeparam>
+        /// <param name="world">The world instance to perform the move in.</param>
+        public static void MoveSystemFromToGroup<System, SrcGroup, DestGroup>(World world)
+                where System : ComponentSystemBase
+                where SrcGroup : ComponentSystemGroup
+                where DestGroup : ComponentSystemGroup
+        {
+            ComponentSystemBase system = world.GetExistingSystem<System>();
+            ComponentSystemGroup srcGroup = world.GetExistingSystem<SrcGroup>();
+            ComponentSystemGroup destGroup = world.GetExistingSystem<DestGroup>();
+
+            // Skip if there are missing elements
+            if (system == null || srcGroup == null || destGroup == null)
+            {
+                throw new ArgumentNullException($"{nameof(MoveSystemFromToGroup)}: One or more of the provided system and groups do not exist. {nameof(System)}:{system}, {nameof(SrcGroup)}:{srcGroup}, {nameof(DestGroup)}:{destGroup}");
+            }
+
```

Comment:
Should we add a check to ensure that `destGroup` and `srcGroup` are not the same?

Reply:
yea, I guess I can debug assert that. Really, it's not a big deal if your source and destination systems are the same. You're just doing work for nothing.

Reply:
fixed

Reply:
My reasoning was that maybe there would be an issue with OnCreated, OnDestroyed, OnStartRunning, OnStopRunning etc being called that we didn't want. 

But you added the assert so all good.